fetch.bs

```diff
@@ -7558,6 +7559,14 @@ dictionary RequestInit {
   RequestDuplex duplex;
   RequestPriority priority;
   any window; // can only be set to null
+  FetchObserverCallback observer;
+};
+
+callback FetchObserverCallback = undefined (FetchObserver requestObserver, FetchObserver responseObserver);
```

Comment:
Firefox has the below IDL. I split it so the observers still fire events named progress but it's maybe fine to keep it as a single eventtarget and just change the event names?

```webidl
[Exposed=Window]
callback interface ObserverCallback {
  undefined handleEvent(FetchObserver observer);
};

enum FetchState {
  // Pending states
  "requesting", "responding",
  // Final states
  "aborted", "errored", "complete"
};

[Exposed=(Window,Worker),
 Pref="dom.fetchObserver.enabled"]
interface FetchObserver : EventTarget {
  readonly attribute FetchState state;

  // Events
  attribute EventHandler onstatechange;
  attribute EventHandler onrequestprogress;
  attribute EventHandler onresponseprogress;
};
```